49.md

```diff
@@ -0,0 +1,83 @@
+
+As well, Nostr Wallet Connect has lots of different functions that a wallet may or may not support or a service may or
+may not need. This either runs into silent failures or requires the user to understand what functions their wallet
+supports and what functions the service needs.
+
+Finally, Nostr Wallet Connect requires the wallet service to generate both key pairs and the service to use these. This
+limits the key management strategies that a service can use and often leads to suboptimal key management.
+
+This NIP proposes a new protocol that solves these problems by having the client and service generate a NWC connection
+URI together. This URI is then used to connect the client and service.
+
+## Theory of Operation
+
+1. The user wants to connect their wallet to a service. The service provides a QR code or other means of connecting to
+   the service. The QR code contains a `nip49://` URI with a `service_key` parameter that contains the service's public
+   key, the relay to communicate over, and a list of functions that the service requires.
+2. The user scans the QR code and their wallet prompts them the required functions and asks them to confirm that they
+   want to connect to the service.
+3. On confirmation, the wallet generates a new key pair and sends the public key to the service in a confirmation event.
+4. With the combination of both keys, the wallet and the service both will generate the same connection URI. They can
+   now use NWC to communicate.
+
+## Specification
+
+### URI Format
+
+The **service** generates this connection URI with protocol `nip49:` and base path it's hex-encoded `pubkey` with the
+following query string parameters:
+
+- `relay` Required. URL of the relay where the **wallet service** is connected and will be listening for events. May be
+  more than one.
+- `required_commands` Required. A space-separated list of commands that the **wallet service** requires. The **client**
+  MUST NOT
+  connect if it does not support all of these permissions.
+- `optional_commands` Optional. A space-separated list of commands that the **wallet service** can enable to add
+  additional functionality. The **client** MAY ignore these.
+- `budget` Optional. A budget that the **wallet service** will use to limit the amount of funds that the **service** can
+  spend. The **service** MUST NOT spend more than this amount. The budget is a string of the form `max_amount/period`.
+  The `period` is one of `day`, `week`, `month`, `year`. The `max_amount` is a satoshi amount.
+
+#### Example URI
+
+```sh
+nostr+walletauth://b889ff5b1513b641e2a139f661a661364979c5beee91842f8f0ef42ab558e9d4?relay=wss%3A%2F%2Frelay.damus.io&required_commands=pay_invoice%20pay_keysend%20make_invoice%20lookup_invoice&optional_commands=list_invoices%20list_payments&budget=10000%2Fday
+```
+
+### Event
+
+This NIP only uses one event kind `13193`. We use a replaceable event to allow for the wallet and service to renegotiate
```

Comment:
Who writes this event? And how does this "negotiation" work? What messages go back and forth? What's the end goal of this negotiation? 

Maybe an example of all messages going back and forth can clarify this